### PR TITLE
20 healing functionality

### PR DESCRIPTION
**Pull request by Mlodko**
# What was added
## Healing functionality
You can now call the `heal()` function in any Hit Box to heal it for a specified amount. 
### Debug
For debugging, you can manually heal the player by hitting space.

## Documentation
- Changed .gitignore to allow obisidian plugins
- Added some code documentation with metadata (tags and properties)

### CODE REVIEW

**Comment by stas420** [comment]
Looks good, code docs will need to be grouped into directories, but it is a thing of future. Code seems good as a basics-setting one.

File: MOT Docs/.obsidian/community-plugins.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[
```
Comment (ddziechciarz) [inline MOT Docs/.obsidian/community-plugins.json:1]: I thought we don't want .obsidian in our repository

File: Assets/Utility/DamageBoxes/hit_box.gd [hunk 2]
```diff
@@ -30,6 +35,13 @@ func take_damage(dmg: Damage) -> void:
 		die()
 	update_health.emit()
 
+func heal(heal_amount : float) -> void:
+	var calculated_healing = heal_amount # Modifiers here
+	health = max_health if health + calculated_healing >= max_health else health + calculated_healing
```
Comment (ddziechciarz) [inline Assets/Utility/DamageBoxes/hit_box.gd:40]: health = min(health + calculated_healing, max_health) is shorter and more readible

Comment (Mlodko) [reply Assets/Utility/DamageBoxes/hit_box.gd:40]: True, but I have a very important reason for doing it this way
I forgor

Comment (Mlodko) [reply Assets/Utility/DamageBoxes/hit_box.gd]: Fixed in 1baf372

**Review by ddziechciarz — APPROVED** [review]
git

**Pull request merged**